### PR TITLE
Keep Track of Skips in Rapid Succession

### LINKED ISSUE #969: Keep Track of Back/Forward Skips in Rapid Succession
### What problem does this address?

If a user skips ahead 30 seconds with a button press, then immediately skips backwards 15 seconds, the "time saved skipping forward" that's recorded is 30 seconds saved. Or, if a user skips forward 30 seconds and immediately skips backwards 30 seconds, that's recorded as 30 seconds saved.

### What is your proposed solution?

We should keep track of backward and forward skips within a two second window in `PlaybackManager.kt` and only send the difference to `statsManager`

### Did you search for existing feature suggestions?

- [X] I have searched for existing feature suggestions.

### PR DESCRIPTION
**Pull request by leoherzog**
## Description
If a user skips ahead 30 seconds with a button press, then immediately skips backwards 15 seconds, the "time saved skipping forward" that's recorded is 30 seconds saved. Or, if a user skips forward 30 seconds and immediately skips backwards 30 seconds, that's recorded as 30 seconds saved.

We should keep track of backward and forward skips within a two second window in `PlaybackManager.kt` and only send the difference to `statsManager`.

Fixes #969 

## Testing Instructions

1. Skip forwards then quick skip backwards
2. Check the stats and make sure that only the difference was sent.

### CODE REVIEW

**Comment by mchowning** [comment]
👋 @leoherzog ! Thanks so much for your contribution!

We're a bit backed up with work right now, but we'll review this as soon as we can.

**Review by mchowning — COMMENTED** [review]
👋 @leoherzog ! Sorry it took so long to review this. 🙇 We really appreciate your contribution!

This change works as you describe and does adjust the stats when someone skips forward and then immediately skips backward a bit. 👍 

I think this change does introduce a new way for the stats to be off though. In particular, if I skip backward more than I skipped forward, I can take my skip forward stat negative (possibly significantly negative if I skip back a long way).

I am feeling a bit uncertain about this PR's change as whole because (1) it adds a bit of code complexity, (2) I'm uncertain how much of an effect #969 is really having on stats, and (3) it introduces a new (although less likely) way for the stats to be off.

I think we could significantly limit the impact the new issue (users skipping back a bunch after they skip forward) by just not resetting the `lastSkipTime` each time they skipped back. That way only a few skip backs would be accounted for.

Regarding how much of an impact #969 is having on stats, I'm wondering if there is a scenario you're thinking of where not keeping track of skips in rapid succession is likely to cause the stats to be significantly wrong? Maybe I'm just not thinking about the right use case where this is going to come up a lot for some users and throw their stats way off.

File: modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt [hunk 1]
```diff
@@ -647,6 +647,8 @@ open class PlaybackManager @Inject constructor(
         }
     }
 
+    var lastSkipTimestamp: Long = 0L
```
[the remark this answers is not in the document]
Comment (Mukuljangir372) [reply modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt:650]: Also, we're exposing the field as public. It should be private. Wdyt?

**Review by Mukuljangir372 — APPROVED** [review]

**Comment by MiSikora** [comment]
Closing the PR as we decided not to add this feature: https://github.com/Automattic/pocket-casts-android/issues/969#issuecomment-2368449397

**Pull request closed without merging**